src/vaults/locked/StrategyVaultV2.sol

```diff
@@ -17,8 +18,13 @@ contract StrategyVaultV2 is StrategyVault, NftGate, HarvestStorage {
     function _deposit(address caller, address receiver, uint256 assets, uint256 shares) internal virtual override {
         _checkNft(receiver);
         if (shares == 0) revert VaultErrors.ZeroShares();
+        uint256 tvl = totalAssets();
+        uint256 allowedAssets = tvl >= tvlCap ? 0 : tvlCap - tvl;
+        assets = Math.min(allowedAssets, assets);
```

Comment:
We should use `MathUpgradeable` since that lib is already included in the contract's bytecode.